client/blocks/app-banner/index.jsx

```diff
@@ -158,50 +148,59 @@ export class AppBanner extends Component {
 			return null;
 		}
 
-		this.loadDismissTimesExperiment();
-
 		const { title, copy } = getAppBannerData( translate, currentSection );
 
 		return (
-			<div className={ classNames( 'app-banner-overlay' ) } ref={ this.preventNotificationsClose }>
-				<Card
-					className={ classNames( 'app-banner', 'is-compact', currentSection ) }
-					ref={ this.preventNotificationsClose }
-				>
-					<TrackComponentView
-						eventName="calypso_mobile_app_banner_impression"
-						eventProperties={ {
-							page: currentSection,
-						} }
-						statGroup="calypso_mobile_app_banner"
-						statName="impression"
-					/>
-					<div className="app-banner__circle is-top-left is-yellow" />
-					<div className="app-banner__circle is-top-right is-blue" />
-					<div className="app-banner__circle is-bottom-right is-red" />
-					<div className="app-banner__text-content">
-						<div className="app-banner__title">
-							<span> { title } </span>
-						</div>
-						<div className="app-banner__copy">
-							<span> { copy } </span>
-						</div>
-					</div>
-					<div className="app-banner__buttons">
-						<Button
-							primary
-							className="app-banner__open-button"
-							onClick={ this.openApp }
-							href={ this.getDeepLink() }
+			<ProvideExperimentData name="calypso_mobileweb_appbanner_frequency_20220128_v1">
+				{ ( isLoading, experimentAssignment ) => {
+					this.experimentIsControl = isLoading ? true : ! experimentAssignment?.variationName;
```

Comment:
As I've mentioned in the second part of [my previous comment](https://github.com/Automattic/wp-calypso/pull/60830#discussion_r804159693), we could use the `isLoading` value to prevent displaying the banner until the experiment assignment has finished loading. 

```suggestion
					if ( isLoading ) { 
					    return null;
					}
```
It's a slight UX tradeoff in order to get cleaner experiment results. If this delay is a concern, we can further reduce it by calling `loadExperimentAssignment( APP_BANNER_EXPERIMENT_NAME )`directly in `client/layout/index.jsx` using the same condition (`shouldDisplayAppBanner`) as we use for evaluating the assignment here.